Aggressive sanitizer

ebering opened the pull request:
In a fancier version should have the whitelist be a list of regex to match against so that self. can work if the subreddit isn't politics but thats too much work.

ebering committed:
Move sanitization to rendering

Domain is link to original, main link is sanitized at the rendering
phase.

ebering committed:
Whitelist sanitizing

Run all posts except those on a whitelist through outline.

ebering committed:
Datify whitelisting

aparent commented:
```
Cannot find variable `sanitizeUrl`

162|     , Html.a [ class "storyLink", anim, style_, href sanitizeUrl post.url, title post.title ] [ text post.title ]
```

Possibly forgot to add this function?

ebering committed:
Move SanitizePost -> SanitizeUrl

As sanitizing is part of rendering. This is probably not the best
engineering but I am not an engineer.

j3parker commented on src/SanitizePost.elm:
```diff
@@ -5,22 +5,19 @@ import Reddit exposing (Post)
-    if post.domain |> paywalled then
-        { post | url = depaywall post.url }
-    else
+    if post.domain |> whitelist then
```
Haha

j3parker commented on src/Main.elm:
```diff
@@ -161,8 +158,8 @@ renderPost post =
     in
-    [ Html.span [ class "ref", anim, style_ ] [ text post.domain ]
-    , Html.a [ class "storyLink", anim, style_, href post.url, title post.title ] [ text post.title ]
+    [ Html.span [ class "ref", anim, style_, href post.url ] [ text post.domain ]
```
I think you'll need to switch this to `Html.a` rather than `Html.span`; neat idea RE: making this the unfiltered one

j3parker replied:
And I was a jerk and made the CSS specific, you'd have to change this to `a` too: https://github.com/j3parker/f6oclock/blob/cd53bce66001caad2578a74e7a00d2332f8d42da/src/styles.css#L16

j3parker commented:
O/w looks cool. Ignore CI and merge when you're ready.

ebering committed:
Have 'ref' be a link to the unfiltered post
